### PR TITLE
Add NavSatFixTranslator and the ability to change CRS per message type

### PR DESCRIPTION
**Pull request by romainreignier**
Thanks a lot of this `qgis_ros` package, it is quite useful to use QGis to see geographic data.

So I have added a `NavSatFixTranslator` to use ROS standard `sensor_msgs/NavSatFix` GNSS receiver messages.

I had to add a new CRS in order to get the points on a correct location on a map, so I have added a `crsName` member to the `Translator` base class.

I had to use dicts to store CRS PROJ4 data and `QgsCoordinateReferenceSystem` objects because the raw PROJ4 string contains spaces and cannot be used in the `uri` passed to `QgsVectorLayer` constructor when subscribing to a topic.

### CODE REVIEW

**Comment by ablakey** [comment]
Thanks for the contribution!

I am not in a position to test this with NavSatFix data.  If you've got a simple testing strategy that's reasonably easy and safe to share, that would be great. Otherwise I will defer to you to decide that this works as intended.  The diff looks good to me.

I think there may be a case that if we handle more than 1 CRS, we should be more generic than hard-coding 2 of them. However, I think `WGS84` and `Simple` should cover _most_ robotics + GIS use cases. This is a problem I'm happy to kick down the road.

I'd love to hear more about what you're doing with QGIS-ROS, if you've got any public links/details to share.

Let me know if you think it's good to merge and I'll do so.

**Comment by romainreignier** [comment]
Ok, thank you for the feedback.

Tomorrow, I might be able to record a trivial rosbag in Gazebo for you to test.
Actually, there may be publicly available rosbags from autonomous driving projects like Autoware.

I am not doing much with QGIS-ROS for now, I work on [phenotyping solutions](https://www.robopec.com/en/products/phenotyping/) and wanted to view positions on a map.
I already use QGIS a bit for other geo data like shapefiles and geotiff.
Furthermore, I have seen the talk at ROSCon and always wanted to try it out one day.

**Comment by romainreignier** [comment]
Hi!

I have used a sample bag from [Autoware ROSBAG Demo](https://gitlab.com/autowarefoundation/autoware.ai/autoware/-/wikis/ROSBAG-Demo):

[sample_moriyama_150324.tar.gz](https://autoware-ai.s3.us-east-2.amazonaws.com/sample_moriyama_150324.tar.gz)

It contains `nmea_msgs/Sentence` that I converted to `sensor_msgs/NavSatFix` with the [nmea_navsat_driver](https://wiki.ros.org/nmea_navsat_driver).

So here it is:
[sample_moriyama_150324_navsatfix.bag.tar.gz](https://github.com/locusrobotics/qgis_ros/files/4558970/sample_moriyama_150324_navsatfix.bag.tar.gz)

And the result in QGIS :


The live topic version can be tested with `rosbag play`.

Note that I could not used lz4 compressed bag directly. It seems to be an issue with Python 3 because I can reproduce it with a simple script inspired from [rosbag tutorial](https://wiki.ros.org/rosbag/Code%20API#Python_API). It works with Python 2 but not with Python 3. This may be caused by `ros-melodic-roslz4` that it targeted for Python 2 on Melodic. So it hould work on Noetic.

Also, I have done my PR against the `master` branch but because I am on Melodic, I have actually used the `ab-handle-empty-module-names` branch so maybe this branch should also merged to `master`.

**Comment by ablakey** [comment]
Thanks for the reminder. That branch should be merged. I'll test this out and then merge both today.

File: src/qgis_ros/core/crs.py [hunk 1]
```diff
@@ -12,5 +12,10 @@
                 +units=m \
                 +no_defs'
 
+PROJ4_WGS84 = '+proj=longlat +datum=WGS84 +no_defs'
 
 simpleCrs = QgsCoordinateReferenceSystem('PROJ4:' + PROJ4_SIMPLE)
+wgs84Crs = QgsCoordinateReferenceSystem('PROJ4:' + PROJ4_WGS84)
+
+proj4CrsDict = {'simple': PROJ4_SIMPLE, 'wgs84': PROJ4_WGS84}
+crsDict = {'simple': simpleCrs, 'wgs84': wgs84Crs}
```
Comment (abencz) [inline src/qgis_ros/core/crs.py]: Needs return at end of file.

**Review by ablakey — APPROVED** [review]
Tested with supplied data (thanks for helping supply that!). LGTM.

**Review by abencz — APPROVED** [review]

File: src/qgis_ros/core/crs.py (hunk 1 above)
Comment (ablakey) [reply src/qgis_ros/core/crs.py]: To confirm, you mean a newline character.  Probably a windows vs. linux or code editor detail. I'll add it back in here.

**Pull request merged**